tools/lang/src/compiler/mod.rs

```diff
@@ -165,6 +214,80 @@ fn make_process_restrictions(
     })
 }
 
+fn map_token_props_to_arguments(token_decl: &TokenDecl) -> Vec<ArgumentField> {
+    token_decl
+        .props
+        .value
+        .iter()
+        .map(|prop| ArgumentField {
+            name: prop.value.name.value.to_owned(),
+            allow_token: prop
+                .value
+                .types
+                .iter()
+                .find(|t| matches!(t.value, TokenFieldType::Token(_)))
+                .is_some(),
+            allow_role: prop
+                .value
+                .types
+                .iter()
+                .find(|t| matches!(t.value, TokenFieldType::Role))
+                .is_some(),
+            allow_file: prop
+                .value
+                .types
+                .iter()
+                .find(|t| matches!(t.value, TokenFieldType::File))
+                .is_some(),
+            allow_literal: prop
+                .value
+                .types
+                .iter()
+                .find(|t| matches!(t.value, TokenFieldType::Literal))
+                .is_some(),
+            allow_none: prop
+                .value
+                .types
+                .iter()
+                .find(|t| matches!(t.value, TokenFieldType::None))
+                .is_some(),
+            allowed_literal_values: prop
+                .value
+                .types
+                .iter()
+                .filter_map(|t| match &t.value {
+                    TokenFieldType::LiteralValue(v) => Some(v.value.to_owned()),
+                    _ => None,
+                })
+                .collect(),
+        })
+        .collect()
+}
+
+fn make_process_arguments(fn_decl: &FnDecl, token_decls: &HashMap<&str, TokenDecl>) -> Arguments {
```

Comment:
note to self this is broken. Inputs should be specified as token ids and for outputs we need to output the token type and version